posts/2021-03-24-whats-new-in-MP-Rest-Client2.0.adoc

```diff
@@ -0,0 +1,201 @@
+instance or via MP Config. Here is an example of configuring auto-redirect via the `RestClientBuilder` API:
+
+[source,java]
+----
+MyClient client = RestClientBuilder.newBuilder()
+                                   .followRedirects(true)
+                                   //...
+                                   .build(MyClient.class);
+----
+
+And here is how you would configure it via MP Config:
+
+`com.mypkg.MyClient/mp-rest/followRedirects=true`
+
+[#sse-support]
+== Support for Server Sent Events (SSEs)
+
+link:https://openliberty.io/guides/reactive-messaging-sse.html[Server Sent Events] are part of the HTML 5 spec and
+enables a server to push data to a client asynchronously via events - all over HTTP. The JAX-RS 2.1 spec enabled SSE
+support for both the client and server. Now you can consume SSE events from the type-safe MP Rest Client.
+
+The MP Rest Client specification uses the link:http://www.reactive-streams.org/[Reactive Streams] APIs to consume
+events. A client interface capable of consuming SSEs would look something like this:
+
+[source,java]
+----
+@RegisterRestClient
+public interface SseClient {
+
+    @GET
+    @Path("/path/sse")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    Publisher<String> getStrings();
+
+    @GET
+    @Path("/path/sse2")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    Publisher<InboundSseEvent> getEvents();
+}
+----
+
+First, the method (or interface) must be annotated with `@Produces(MediaType.SERVER_SENT_EVENTS)` to indicate that it
+expects the server to produce SSEs. Next, the method's return type must be `org.reactivestreams.Publisher`. The generic
+type can be `javax.ws.rs.sse.InboundSseEvent` from JAX-RS or it can be a primitive or complex object - the latter only
+works if (1) the server only sends one type of events (e.g. only sends `WeatherEvents` - then `Publisher<WeatherEvent>`
+would be applicable) and (2) if you have an entity provider that is capable of converting the events into the generic
+type (e.g. `MessageBodyReader<WeatherEvent>`). For most cases, if the remote server sends events using JSON, you can
+enable the `jsonb-1.0` feature in your Liberty server which will automatically register a JSON-B-based entity provider.
+
+Once you've invoked one of these methods, you should be able to register one or more `Subscriber` instances to the
+`Publisher`. Each subscriber will be notified on receipt of a new event or if the connection to the server has been
+closed.
+
+== Summary
+
+MicroProfile Rest Client 2.0 has some powerful new features useful for building cloud native applications. There are
+still more features not covered here, but have been in Open Liberty's previous Rest Client support, including using CDI-
+enabled providers and supporting `ContextResolver` to customize the built-in JSON-B provider. These features and more
+are available on the link:https://github.com/eclipse/microprofile-rest-client/releases/tag/2.0[MP Rest Client 2.0 release page].
```

Comment:
Does "JSON-B ContextResolver Support" not deserve it's own section? If not then it seems a little confusing to mention it here.
```suggestion
MicroProfile Rest Client 2.0 has some powerful new features useful for building cloud native applications. You can read more about these updates on the link:https://github.com/eclipse/microprofile-rest-client/releases/tag/2.0[MP Rest Client 2.0 release page].
```